### PR TITLE
400 create citations using GitHub

### LINKED ISSUE #400: Create citations using GitHub
**Is your feature request related to a problem? Please describe.**
I think it would be beneficial to use citations in MAPIE (directly from the GitHub features). Checkout [link](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files).

### PR DESCRIPTION
**Pull request by LacombeLouis**
# Description
This pull request introduces the addition of citation details to the MAPIE project by including a `CITATIONS.cff` file. The goal is to facilitate proper citation of MAPIE in academic and professional work by providing a standardized citation format. This change addresses the issue raised regarding the beneficial use of citations directly from the GitHub features. The `CITATIONS.cff` file includes author details, software title, version, release date, URL, and a preferred citation format.

Fixes #400.

## Type of change:
Please remove options that are irrelevant.
- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Verified that the `CITATIONS.cff` file is properly formatted and recognized by GitHub
- [x] Ensured the citation information is correct and matches the required format

# Checklist

- [x] I have read the [contributing guidelines](https://github.com/scikit-learn-contrib/MAPIE/blob/master/CONTRIBUTING.rst)
- [x] I have updated the [HISTORY.rst](https://github.com/scikit-learn-contrib/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/scikit-learn-contrib/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `make lint`
- [x] Typing passes successfully : `make type-check`
- [x] Unit tests pass successfully : `make tests`
- [x] Coverage is 100% : `make coverage`
- [x] Documentation builds successfully : `make doc`

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    given-names: "Vincent"
+  - family-names: "Morzadec"
+    given-names: "Thomas"
+  - family-names: "Lacombe"
+    given-names: "Louis"
+  - family-names: "Brunel"
+    given-names: "Nicolas"
+  doi: "10.48550/arXiv.2207.12274"
+  journal: "arXiv preprint arXiv:2207.12274"
+  title: "MAPIE: an open-source library for distribution-free uncertainty quantification"
```
Comment (thibaultcordier) [inline CITATION.cff:26]: Is it possible to quote the paper presented to COPA (as a replacement or in addition)?

**Review by thibaultcordier — COMMENTED** [review]
Thank you for this PR, which will enable the project to be more widely cited.

Is it also possible to add a citation at the end of the readme? (https://github.com/scikit-learn/scikit-learn?tab=readme-ov-file#citation)

File: CITATION.cff (hunk 1 above)
Comment (LacombeLouis) [reply CITATION.cff:26]: Could you remind me the link to this paper? My only concern is that this paper also presented a new insight and was not a general overview of the library.

What do you think @vincentblot28 ?

Comment (vincentblot28) [reply CITATION.cff:26]: I think we must add the COPA paper as we also present MAPIE in it. https://proceedings.mlr.press/v204/cordier23a/cordier23a.pdf

File: README.rst [hunk 2]
```diff
@@ -224,3 +224,9 @@ and with the financial support from Région Ile de France and Confiance.ai.
 ==========
 
 MAPIE is free and open-source software licensed under the `3-clause BSD license <https://github.com/simai-ml/MAPIE/blob/master/LICENSE>`_.
+
+
+📚 Citation
+===========
+
+If you use MAPIE in your research, please cite using `citations file <https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files>`_ on our repository.
```
Comment (thibaultcordier) [inline README.rst:232]:
```suggestion
If you use MAPIE in your research, please cite using `citations file <CITATION.CFF>`_ on our repository.
```

Comment (thibaultcordier) [reply README.rst:232]: Check that this change works.

File: README.rst [hunk 3]
```diff
@@ -224,3 +224,9 @@ and with the financial support from Région Ile de France and Confiance.ai.
 ==========
 
 MAPIE is free and open-source software licensed under the `3-clause BSD license <https://github.com/simai-ml/MAPIE/blob/master/LICENSE>`_.
```
Comment (thibaultcordier) [inline README.rst:226]:
```suggestion
MAPIE is free and open-source software licensed under the `3-clause BSD license <LICENSE>`_.
```

Comment (thibaultcordier) [reply README.rst:226]: Check that this change works.

File: README.rst
Comment (thibaultcordier) [inline README.rst:1]: Change file path names:

- l15: https://img.shields.io/github/license/simai-ml/MAPIE -> https://img.shields.io/github/license/scikit-learn-contrib/MAPIE
- l63: simai-ml -> scikit-learn-contrib
- l161: <https://github.com/simai-ml/MAPIE/issues> -> <https://github.com/scikit-learn-contrib/MAPIE/issues>

**Review by thibaultcordier — CHANGES_REQUESTED** [review]
Thank you for these changes. Once again, these are just a few harmonised changes. At the same time, I don't understand why we don't see all the citation information in the "Cite the repository" function...

File: README.rst (hunk 2 above)
Comment (LacombeLouis) [reply README.rst]: I'm not sure this is useful as it does not tell you how to use the `CITATION.CFF file`.
But we can link to that anyways.

File: README.rst
Comment (LacombeLouis) [reply README.rst:1]: done

**Pull request merged**